Code/Framework/AzFramework/AzFramework/Script/ScriptComponent.cpp

```diff
@@ -564,6 +564,11 @@ namespace AzFramework
         // if we have valid asset listen for script asset events, like reload
         if (m_script.GetId().IsValid())
         {
+            AZ::Data::AssetLoadParameters loadParams;
+            m_script = AZ::Data::AssetManager::Instance().GetAsset<AZ::ScriptAsset>(
+                    m_script.GetId(),
+                    m_script.GetAutoLoadBehavior(),
+                    loadParams);
             AZ::Data::AssetBus::Handler::BusConnect(m_script.GetId());
             m_script.QueueLoad();
```

Comment:
there's currently a bug within asset system which releases m_script asset meanwhile keeping it cached on prefab. It causes the following issue:
1. Start fresh instance of Editor
2. Create a prefab with Lua script attached.
3. Run Game Mode and spawn this prefab whichever way possible (Lua, SC, or C++)
4. Exit Game Mode.
5. Make any change to Lua script in step 2 and save it.
6. Run game mode again
7. The Lua script on spawned entity will no longer work

This is because it gets released by asset manager (in OnAssetReloaded) as it doesn't account for the asset reference of the script attached to the prefab. However when we run Game Mode again, and spawn the same prefab, its script asset is also cloned and is now in 'Ready' state (but missing from AssetManager!). So it never gets properly loaded and never gets activated. I have talked with @lawsonamzn and @carlitosan and this was the only possible temp fix for this issue.

Reply:
added comment for clarity

Reply:
@AMZN-mnaumov , can you create a github issue for this if one doesn't exist already?

Reply:
https://github.com/o3de/o3de/issues/8760